### DISCUSSION TITLE
Dropdown content not taking the full width of the trigger

### DISCUSSION THREAD

**Discussion by raviteja83** [radix-ui/primitives#1507] [Help] [answered]
Are there any limitations for the width the dropdown content can take? The content is not taking the full width of trigger. Anything that can be done? I have set no width constraint on the dropdown content.

<img width="569" alt="image" src="https://user-images.githubusercontent.com/6763261/177194393-665500bb-0785-4acb-a577-78f039454b6a.png">

**Comment by benoitgrelard** [comment, marked as answer]
Hi @raviteja83,

There are no "limitations", but no constraints either.
The content width will simply be the natural width, so it will depend on the width of your larger item, paddings, etc, like a regular element.

If you wanted to make them match, you could either do it with css, by applying the same width to the trigger and the content, or dynamically by not applying a specific width to the trigger, but rather measuring it and applying that measured width to the content.

Hope this helps.

✌️

**Reply by raviteja83** [reply]
@benoitgrelard is there a way I can set the width of menu to be 100% of the parent where the trigger is rendered? Setting a measured width works but I want to avoid this.

**Reply by benoitgrelard** [reply]
You can do this potentially if you don't portal the content, currently `portalled={false}` (or not rendering `DropdownMenu.Portal` if using the latest release candidate). This will make the content render next to the trigger.
You potentially lose control over some tricky layering z-index scenarios, but it might work for you otherwise.

**Reply by raviteja83** [reply]
Thanks for the quick response. i am trying the same but the alignment of the menu is off. Am I missing anything? Might be positioning is not happening with respect to root or trigger.

<img width="670" alt="image" src="https://user-images.githubusercontent.com/6763261/177303196-4f50cd6e-6bb0-4296-a9be-4b1d0d00eeef.png">

**Reply by benoitgrelard** [reply]
Looks like you are hitting #781 so you might not be able to do it this way at the moment unfortunately.

Btw, I just realised that you are using the wrong component for your use-case.
You should be using `Select` as opposed to `DropdownMenu`.
`DropdownMenu` is to be used for a menu of actions, not to maintain a selected value, it is not the right semantic for it.

`Select` will also have the behaviour you are after in terms of the matching size, or at least closer I think.

**Reply by raviteja83** [reply]
The actual idea was to use select itself but because of https://github.com/radix-ui/primitives/issues/1247 I had to try to do with dropdown.

**Reply by benoitgrelard** [reply]
Yeah, the positioning is a limitation at the moment, that being said, by using the wrong component, you are using the wrong semantic, making it worse accessibility-wise, as opposed to something that is correct semantically, accessible but looks slightly different to what you want (for now).

**Reply by raviteja83** [reply]
#781 helped me for now. Thanks for the quick response. Great work on the library. Thank you very much.

**Comment by flaviodelgrosso** [comment]
> Are there any limitations for the width the dropdown content can take? The content is not taking the full width of trigger. Anything that can be done? I have set no width constraint on the dropdown content.
> 
> <img alt="image" width="569" src="https://user-images.githubusercontent.com/6763261/177194393-665500bb-0785-4acb-a577-78f039454b6a.png">

In the className attribute of `DropdownMenuContent` you can set this: `w-[--radix-dropdown-menu-trigger-width]` (that's tailwind based but the concept is the same)

**Reply by Itzadetunji** [reply]
This should be the updated class  w-[--radix-popper-anchor-width]

**Comment by AdiCahyaSaputra** [comment]
Hey take a look at this

https://www.radix-ui.com/primitives/docs/components/dropdown-menu#constrain-the-contentsub-content-size

**Comment by ChoaibMouhrach** [comment]
style={{ width: "var(--radix-popper-anchor-width)" }}

**Comment by connecttoMAHDI** [comment]
Hello, today I faced the same challenge and the fix was straight forward:
Radix creates these CSS variables automatically:
--radix-dropdown-menu-trigger-width
--radix-dropdown-menu-trigger-height

So the only thing you have to do is to add this to the DropdownMenuContent className:
w-(--radix-dropdown-menu-trigger-width)
